pkg/controller/controller.go

```diff
@@ -427,3 +434,52 @@ func (c *Controller) collectConfigMap() error {
 
 	return nil
 }
+func cloneFunction(in interface{}) (*spec.Function, error) {
+	var scheme = runtime.NewScheme()
+	clone, err := scheme.DeepCopy(in)
+	if err != nil {
+		return nil, err
+	}
+
+	out, ok := clone.(*spec.Function)
+	if !ok {
+		return nil, fmt.Errorf("unexpected type: %T", clone)
+	}
+
+	return out, nil
+}
+
+func (c *Controller) updateFunctionStatus(funcObj *spec.Function) error {
+
+	funcObjCopy, err := cloneFunction(funcObj)
+	if err != nil {
+		return err
+	}
+
+	status := spec.FunctionStatus{Status: spec.NotReadyPhase}
+
+	// TODO: We need a sync loop to periodically check if actual state and
+	// desired state are as expected. Refect the status as per the check.
+	ticker := time.NewTicker(time.Millisecond * 500)
```

Comment:
I don't think we should go ahead with this solution if this doesn't have a watcher. If the function takes more than 3 seconds to deploy or if it crashes after that the status will be wrong (and it is not going to be updated).

Reply:
See TODO note above. We need to add control loop that periodically verifies difference in desired state and actual state. See #184. Its common pattern in controller to do periodic sync independent of the updates from the API server.

Controller is source of truth, and need to reflect the status.

Reply:
yes, what I mean is that we should implement the TODO before merging this PR (I am not sure if you planned to implement it here or in a different PR)

Reply:
I am trying to put a structure to the things step by step in small incremental PR's. Things have to evolve. This is first step. Next we need to define what are life-cycle phases/stages of the function. Also old way of representing `actual/old deployment count` of pods need a place holder in the status with more meaningful name.

I will send out subsequnet PR's

Reply:
Opened https://github.com/kubeless/kubeless/issues/496 for tracking.

Reply:
thanks, I would wait for that to be ready (you can base the changes of that Issue on this branch) so we can merge them together. I would like to avoid if possible a "non-releasable" status of the master branch.

Reply:
Polling every half second is very expensive.  Can we just `watch` the relevant objects instead?

Reply:
As discussed in #184, i will add periodic sync loop that should update the function status as soon deployment is as expected.